### DISCUSSION TITLE
Using functions in channel encodings

### DISCUSSION THREAD

**Discussion by marr** [uwdata/mosaic#134] [Q&A] [answered]
Is there a way to use functions like so when defining channel?

```
vg.dot(vg.from('stocks'), {
  x: (d) => (d.Close - d.Open) / d.Open,
  y: 'Volume',
  r: 'Volume'
}),
```

My plot does not appear because in the select statement I see in the log, it is only pulling the `Volume` column.

**Comment by jheer** [comment, marked as answer]
Mosaic pushes computation to the database, so instead of a JS function you should be able to provide a SQL expression.

``` js
vg.dot(vg.from('stocks'), {
  x: vg.sql`(Close - Open) / Open`,
  y: 'Volume',
  r: 'Volume'
}),
```

Depending on the column types you may also want to include type casts where needed.

**Reply by marr** [reply]
That's great. Thank you Jeffrey!
